src/window_actorstatus.cpp

```diff
@@ -45,24 +45,26 @@ void Window_ActorStatus::DrawStatus(){
 
 	// Draw Hp
 	contents->TextDraw(1, 2, 1, lcf::Data::terms.health_points);
-	DrawMinMax(90,2,actor->GetHp(), actor->GetMaxHp());
+	int color = (actor->GetHp() == 0 ? Font::ColorKnockout : (actor->GetHp() <= actor->GetMaxHp() / 4 ? Font::ColorCritical : Font::ColorDefault));
```

Comment:
Double ternary is not easy to read IMO. This could be changed to a function lambda I guess, since almost the same calculation is done for HP and SP.

Reply:
Please excuse my dumb question, but what does "function lambda" mean?

Reply:
untested proposal:

```c++
auto get_fontcolor() = [](int have, int max, bool can_knockout) {

	if (can_knockout && have == 0) return Font::ColorKnockout;

	if (max > 0 && (have <= max / 4)) return Font::ColorCritical;

	return Font::ColorDefault;

}
```

Use like this:

```c++
get_fontcolor(actor->GetHp(), actor->GetMaxHp(), true)
```
```c++
get_fontcolor(actor->GetSp(), actor->GetMaxSp(), false)
```

What I dislike here is that the arguments are duplicated in the `DrawMinMax()` call then.

Reply:
maybe even with captures:

```diff
diff --git a/src/window_actorstatus.cpp b/src/window_actorstatus.cpp
index b716b994..95b4c640 100644
--- a/src/window_actorstatus.cpp
+++ b/src/window_actorstatus.cpp
@@ -43,15 +43,24 @@ void Window_ActorStatus::DrawStatus() {
 
 	Game_Actor* actor = Main_Data::game_actors->GetActor(actor_id);
 
+	int have, max;
+	auto fontcolor = [&have, &max](bool can_knockout) {
+		if (can_knockout && have == 0) return Font::ColorKnockout;
+		if (max > 0 && (have <= max / 4)) return Font::ColorCritical;
+		return Font::ColorDefault;
+	};
+
 	// Draw Hp
 	contents->TextDraw(1, 2, 1, lcf::Data::terms.health_points);
-	int color = (actor->GetHp() == 0 ? Font::ColorKnockout : (actor->GetHp() <= actor->GetMaxHp() / 4 ? Font::ColorCritical : Font::ColorDefault));
-	DrawMinMax(90, 2, actor->GetHp(), actor->GetMaxHp(), color);
+	have = actor->GetHp();
+	max = actor->GetMaxHp();
+	DrawMinMax(90, 2, have, max, fontcolor(true));
 
 	// Draw Sp
 	contents->TextDraw(1, 18, 1, lcf::Data::terms.spirit_points);
-	color = (actor->GetMaxSp() > 0 && actor->GetSp() <= actor->GetMaxSp() / 4 ? Font::ColorCritical : Font::ColorDefault);
-	DrawMinMax(90, 18, actor->GetSp(), actor->GetMaxSp(), color);
+	have = actor->GetSp();
+	max = actor->GetMaxSp();
+	DrawMinMax(90, 18, have, max, fontcolor(false));
 
 	// Draw Exp
 	contents->TextDraw(1, 34, 1, lcf::Data::terms.exp_short);
```

Reply:
Your proposal works for me. I have added an extra commit which contains these changes.